### PR TITLE
LUCENE-9618: Do not call IntervalIterator.nextInterval after NO_MORE_DOCS returned

### PR DESCRIPTION
**Pull request by zhaih**
# Description

 * In `ConjunctionIntervalIterator` check whether approximation's returned docId is NO_MORE_DOCS to avoid `nextInterval()` call after NO_MORE_DOCS is returned
 * Add a test case to verify the problem is addressed

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalIterator.java [hunk 1]
```diff
@@ -82,6 +82,11 @@ public int width() {
   /**
    * Advance the iterator to the next interval
    *
+   * Should not be called after {@link DocIdSetIterator#NO_MORE_DOCS} is returned by other methods
```
Comment (mikemccand) [inline lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalIterator.java:85]: Hmm maybe be more specific than `other methods`?  E.g. maybe say `returned by the query scorer's nextDoc() method`?

File: lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalIterator.java [hunk 2]
```diff
@@ -82,6 +82,11 @@ public int width() {
...
+   * if that's the case in some existing code, please consider opening an issue
```
Comment (mikemccand) [inline lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalIterator.java:86]: This is a new sentence -- maybe add `.` at end of previous one and capitlize `If`?

File: lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalIterator.java [hunk 3]
```diff
@@ -82,6 +82,11 @@ public int width() {
...
+   * However, after {@link IntervalIterator#NO_MORE_INTERVALS} is returned by this method, it might be
+   * called again
```
Comment (mikemccand) [inline lucene/queries/src/java/org/apache/lucene/queries/intervals/IntervalIterator.java:88]: Period at end of this sentence?

File: lucene/CHANGES.txt [hunk 4]
```diff
@@ -84,6 +84,8 @@ API Changes
 
 Improvements
 
+* LUCENE-9618: Do not call IntervalIterator.nextInterval after NO_MORE_DOCS is returned
```
Comment (mikemccand) [inline lucene/CHANGES.txt:87]: You can add your name to the end of this `CHANGES` entry (in parens)?

**Review by mikemccand — APPROVED** [review]
Thank you @zhaih -- nice catch, nice test case!

File: lucene/CHANGES.txt (hunk 4 above)
Comment (zhaih) [reply lucene/CHANGES.txt]: Thanks for reminding me! Done

**Pull request merged**